Content.Client/Administration/BwoinkSystem.cs

```diff
@@ -16,46 +21,87 @@ namespace Content.Client.Administration
 
         protected override void OnBwoinkTextMessage(BwoinkTextMessage message, EntitySessionEventArgs eventArgs)
         {
             base.OnBwoinkTextMessage(message, eventArgs);
             LogBwoink(message);
             // Actual line
-            var window = EnsureWindow(message.ChannelId);
-            window.ReceiveLine(message.Text);
+            var window = EnsurePanel(message.ChannelId);
+            window.ReceiveLine($"[color=gray]{message.SentAt.ToShortTimeString()}[/color] {message.Text}");
             // Play a sound if we didn't send it
             var localPlayer = _playerManager.LocalPlayer;
             if (localPlayer?.UserId != message.TrueSender)
             {
                 SoundSystem.Play(Filter.Local(), "/Audio/Effects/adminhelp.ogg");
                 _clyde.RequestWindowAttention();
             }
+
+            _adminWindow?.ChannelSelector?.Refresh();
         }
 
-        public BwoinkWindow EnsureWindow(NetUserId channelId)
+        public bool TryGetChannel(NetUserId ch, [NotNullWhen(true)] out BwoinkPanel bp) => _activePanelMap.TryGetValue(ch, out bp!);
+
+        private BwoinkPanel EnsureAdmin(NetUserId channelId)
         {
-            if (!_activeWindowMap.TryGetValue(channelId, out var existingWindow))
+            if (_adminWindow is null)
+            {
+                _adminWindow = new BwoinkWindow(this);
+            }
+
+            if (!_activePanelMap.TryGetValue(channelId, out var existingPanel))
             {
-                _activeWindowMap[channelId] = existingWindow = new BwoinkWindow(channelId,
-                    _playerManager.SessionsDict.TryGetValue(channelId, out var otherSession)
-                        ? otherSession.Name
-                        : channelId.ToString());
+                _playerManager.SessionsDict.TryGetValue(channelId, out var otherSession);
+                existingPanel = new BwoinkPanel(this, channelId);
+                _activePanelMap[channelId] = existingPanel;
+            }
+
+            if (!_adminWindow.IsOpen)
+            {
+                _adminWindow.Open();
+
+                var csl = _adminWindow.ChannelSelector.PlayerItemList;
+                csl.ClearSelected();
+
+                var pi = csl.FirstOrDefault(i => ((PlayerInfo) i.Metadata!).SessionId == channelId);
+                if (pi is not null)
+                    pi.Selected = true;
             }
+            return existingPanel;
+        }
+
+        private BwoinkPanel EnsurePlain(NetUserId channelId)
+        {
+            if (_plainWindow is null)
+            {
+                var bp = new BwoinkPanel(this, channelId);
+                _plainWindow = new BwoinkedWindow(bp);
+            }
+
+            _plainWindow.Open();
+            return _plainWindow.Bwoink!;
+        }
+
+        public BwoinkPanel EnsurePanel(NetUserId channelId)
+        {
+            if (_adminManager.HasFlag(AdminFlags.Adminhelp))
```

Comment:
should also check if the admin is active using `_adminManager.IsActive()`

Reply:
[No.](https://github.com/space-wizards/space-station-14/blob/f186ffe571aefc7b09a6e9c20fb4281bd48e4680/Content.Shared/Administration/AdminData.cs#L25-L33)

Reply:
mfw i checked if the systemmethod checks for active, but not if the admindata method does. mb